slash/core/result.py

```diff
@@ -49,6 +49,40 @@ def _fact_set_callback(self, fact_name, fact_value):
     def is_global_result(self):
         return False
 
+    def serialize(self):
```

Comment:
@maormarcus serialize is currently very verbose and additional fields will likely get forgotten. Perhaps consider a serialization/deserialization library supporting "exceptions" (enabling us to only conditionally skip data and details?)

I would give marshmallow a try to see if it fits this need